programs/fileio.c

```diff
@@ -796,6 +789,35 @@ static void FIO_adjustMemLimitForPatchFromMode(FIO_prefs_t* const prefs,
     FIO_setMemLimit(prefs, (unsigned)maxSize);
 }
 
+/* FIO_removeMultiFilesWarning() :
+ * Returns 1 if the console should abort, 0 if console should proceed.
+ * Based on displayLevelCutoff (typically == 1) and the global setting g_display_prefs.displayLevel, this
+ * function may print a warning message, a user prompt, both, or none.
+ */
+static int FIO_removeMultiFilesWarning(const FIO_prefs_t* const prefs, int displayLevelCutoff, const char* outFileName)
+{
+    int error;
+    error = 0;
+    if (prefs->nbFiles > 1 && !prefs->overwrite) {
+        if (g_display_prefs.displayLevel <= displayLevelCutoff) {
+            if (prefs->removeSrcFile) {
+                DISPLAYLEVEL(1, "zstd: Aborting... not deleting files and processing into dst: %s", outFileName);
+                error =  1;
+            }
+        } else {
+            if (!strcmp(outFileName, stdoutmark)) {
+                DISPLAYLEVEL(2, "zstd: WARNING: all input files will be processed and concatenated into stdout. ");
```

Comment:
The message is informational and correct : it tells what the action is doing.

But it doesn't tell what are the potential consequences, aka the concatenated output _cannot regenerate the original directory tree_.

This matters particularly in combination with `--rm`, because `zstd -r --rm * -o archive.zst` will simply destroy all information below current directory, with no way to regenerate that from `archive.zst`.
While there might be some valid use cases to do so,
there might also be some cases where some user just expects `archive.zst` to behave about the same as `archive.tar.zst`, the current warning message is not explicit enough to explain that this is a one-way transform with no way back possible.

Probably worth respelling.

Reply:
The like you added `if (prefs->removeSrcFile)` looks good to me. But I think we should always print that line, regardless of `--rm`. Since the user may think they have a full archive and manually delete the directory, for example